src/hooks/useMixpanel.ts

```diff
@@ -0,0 +1,24 @@
+// eslint-disable-next-line no-restricted-imports
+import mixpanel, { Dict, RequestOptions, Callback } from "mixpanel-browser";
+import { useMemo } from "react";
+
+// https://stackoverflow.com/questions/32203420/check-if-mixpanel-library-has-been-loaded
+const isLoaded = () => mixpanel.hasOwnProperty("get_distinct_id");
+
+const noopTrack: (
+  event_name: string,
+  properties?: Dict,
+  optionsOrCallback?: RequestOptions | Callback,
+  callback?: Callback
+) => void = () => {};
```

Comment:
Is there a type exported from mixpanel that we can use for this? Maybe could even do (untested):
```suggestion
const noopTrack: typeof mixpanel.track = () => {};
```